cmd/support-scheduler/main.go

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright (c) 2018 Tencent
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/edgexfoundry/edgex-go"
+	"github.com/edgexfoundry/edgex-go/pkg/config"
+	"github.com/edgexfoundry/edgex-go/pkg/heartbeat"
+	"github.com/edgexfoundry/edgex-go/support/logging-client"
+	"github.com/edgexfoundry/edgex-go/support/scheduler"
+	sc "github.com/edgexfoundry/edgex-go/support/scheduler-client"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+var loggingClient logger.LoggingClient
+
+func main() {
+	start := time.Now()
+	var (
+		useConsul = flag.String("consul", "", "Should the service use consul?")
+		useProfile = flag.String("profile", "default", "Specify a profile other than default.")
+	)
+	flag.Parse()
+
+	//Read Configuration
+	configuration := &scheduler.ConfigurationStruct{}
+	err := config.LoadFromFile(*useProfile, configuration)
+	if err != nil {
+		logBeforeTermination(err)
+		return
+	}
+
+	//Determine if configuration should be overridden from Consul
+	var consulMsg string
+	if *useConsul == "y" {
+		consulMsg = "loading configuration from Consul..."
+		err := scheduler.ConnectToConsul(*configuration)
+		if err != nil {
+			logBeforeTermination(err)
+			return //end program since user explicitly told us to use Consul.
+		}
+	} else {
+		consulMsg = "bypassing Consul configuration..."
+	}
+
+	// Setup Logging
+	logTarget := setLoggingTarget(*configuration)
+	var loggingClient = logger.NewClient(configuration.ApplicationName, configuration.EnableRemoteLogging, logTarget)
+
+	loggingClient.Info(consulMsg)
+	loggingClient.Info(fmt.Sprintf("starting %s %s ", scheduler.SUPPORTSCHEDULERSERVICENAME, edgex.Version))
```

Comment:
`SUPPORTSCHEDULERSERVICENAME` looks like ugly variable name to me.

Reply:
I also think this variable name is terrible, I refered to the other modules. It would be renamed soon.